src/main/java/com/health/utility/ServiceUtility.java

```diff
@@ -449,6 +459,188 @@ public static String presentDirectory() {
+
+                if (!file.exists()) {
+                    logger.info("Sciptfile not found: {}" + file);
+                    continue;
+                }
+
+                try (FileInputStream fis = new FileInputStream(file)) {
+                    zos.putNextEntry(new ZipEntry(file.getName()));
+
+                    int length;
+                    while ((length = fis.read(buffer)) > 0) {
+                        zos.write(buffer, 0, length);
+                    }
+
+                    zos.closeEntry();
+                }
+            }
+
+            String temp = zipFilePath.toString();
+            int indexToStart = temp.indexOf("Media");
+            document = temp.substring(indexToStart, temp.length());
+
+        } catch (Exception e) {
+            logger.error("Exception Error", e);
+        }
+
+        return document;
+    }
+
+    /****
+     * Create ZipFileWithSubDirectories
+     * 
+     */
+
+    private static void zipFile(File fileToZip, String fileName, ZipOutputStream zipOut) throws IOException {
+        if (fileToZip.isHidden()) {
+            return;
+        }
+        if (fileToZip.isDirectory()) {
+            if (fileName.endsWith("/")) {
+                zipOut.putNextEntry(new ZipEntry(fileName));
+                zipOut.closeEntry();
+            } else {
+                zipOut.putNextEntry(new ZipEntry(fileName + "/"));
+                zipOut.closeEntry();
+            }
+            File[] children = fileToZip.listFiles();
+            for (File childFile : children) {
+                zipFile(childFile, fileName + "/" + childFile.getName(), zipOut);
+            }
+            return;
+        }
+        FileInputStream fis = new FileInputStream(fileToZip);
+        ZipEntry zipEntry = new ZipEntry(fileName);
+        zipOut.putNextEntry(zipEntry);
+        byte[] bytes = new byte[16384];
+        int length;
+        while ((length = fis.read(bytes)) >= 0) {
+            zipOut.write(bytes, 0, length);
+        }
+        fis.close();
+    }
+
+    public static String zipFileWithSubDirectories(String sourceDirurl, Environment env) throws IOException {
+
+        String document = "";
+        Path zipFilePathDirectory = Paths.get(env.getProperty("spring.applicationexternalPath.name"),
+                CommonData.uploadDirectoryScriptZipFiles);
+
+        Files.createDirectories(zipFilePathDirectory);
+
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd'T'HHmmss");
+        String zipFileName = "scripts-" + sdf.format(new Date()) + ".zip";
+
+        Path zipFilePathName = Paths.get(env.getProperty("spring.applicationexternalPath.name"),
+                CommonData.uploadDirectoryScriptZipFiles, zipFileName);
+
+        Path sourceDirlName = Paths.get(env.getProperty("spring.applicationexternalPath.name"), sourceDirurl);
+
+        try {
+            OutputStream fos = Files.newOutputStream(zipFilePathName);
+            ZipOutputStream zos = new ZipOutputStream(fos);
```

Comment:
```suggestion
        try (OutputStream fos = Files.newOutputStream(zipFilePathName);
               ZipOutputStream zos = new ZipOutputStream(fos))
        {
```
Use file-with-resources.